### PR TITLE
Guruc 134 edges that disconnect graph

### PR DESCRIPTION
**Pull request by guruc-134**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
 
added the code for a problem related to graphs, the problem uses tarjans algorithm to find the edges which upon removal increase the number of components in the graph i.e (disconnect the graph)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue: #2338

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Why does it look like a solution for a problem from some Competitive Coding site?

**Comment by guruc-134** [comment]
@HarshCasper  the reason is that this is a sum from the category of competitive programming , and the solution is mine you can verify it from [here](https://binarysearch.com/problems/Edges-that-Disconnect-the-Graph/editorials/3980208). I am the author of the solution .
Thanks for reviewing, 😃 shall i change anything in the code? I can make those changes and update the solution

**Comment by guruc-134** [comment]
@HarshCasper  where can I see the changes that are requested, once could you point out where i need to make a change

File: Python/cp/edges_that_disconnect_graph.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# -->parent array contains the parent for each node
+# --> bridges contains the edge(vertex-pairs) which are the bridges
+#  [ not necessary for the question , can be used to understand the answer better]
+# -->time this is the variable which contains the seenAt time which increases
+# by one for each recursive call
+# Implementation:
+# The implementation contains the use of the Trajan's Algorithm.
+# along with some modifications made to accommodate the the conditions
+# --> parent child edge exclusion
+# --> back-edge detection
```
Comment (HarshCasper) [inline Python/cp/edges_that_disconnect_graph.py:87]: Try to ensure that the intuition is captured from the code itself. Here the description is larger than the Code itself and we are not writing articles here

File: Python/cp/edges_that_disconnect_graph.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                # observing for any backedges
+                if lowest_link[v] > seenAt[u] :
+                    bridges.append((u, v))
+            elif v != parent[u]:
+                lowest_link[u] = min(lowest_link[u], seenAt[v])
+    for i in range(n):
+        if seenAt[i] == -1:
+            # call the tarjans() for each of the unvisited node
+            tarjans(i, 0)
+    return len(bridges)
```
Comment (HarshCasper) [inline Python/cp/edges_that_disconnect_graph.py:53]: Add a driver code along with sample I/O and code complexity

File: Python/cp/edges_that_disconnect_graph.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#  GG = [
+#     [1, 2, 3, 5],
+#     [0],
+#     [0, 3],
+#     [0, 2, 4],
+#     [3],
+#     [0]
+# ]
+# output:
+# 3
```
Comment (HarshCasper) [inline Python/cp/edges_that_disconnect_graph.py:15]: Use multi-line strings

File: Python/README.md [hunk 4]
```diff
@@ -54,6 +54,7 @@
 - [Prime factorization using sieve](cp/Prime_factorisation_using_sieve.py)
 - [Stable Marriage Problem](cp/stable_marriage_problem.py)
 - [House Robber](cp/house_robber.py)
+- [edges that disconnect the graph](cp/edges_that_disconnect_graph.py)
```
Comment (HarshCasper) [inline Python/README.md:57]: Make it `Edges that disconnect the Graph`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by guruc-134** [comment]
@HarshCasper  sure, thanks for  the review

**Comment by HarshCasper** [comment]
You don't seem to have addressed any of the reviews. Please close the PR if you are not interested in implementing the changes.

**Pull request closed without merging**